projects/ngrx.io/content/guide/effects/testing.md

```diff
@@ -1,300 +1,258 @@
 
-describe('My Effects', () => {
-  let effects: MyEffects;
-  let actions: ReplaySubject&lt;any&gt;;
+<div class="alert is-helpful">
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [
-        // any modules needed
-      ],
-      providers: [
-        MyEffects,
-        provideMockActions(() => actions),
-        // other providers
-      ],
-    });
+For a detailed look on the marble syntax, see [Writing marble tests](https://github.com/ReactiveX/rxjs/blob/master/doc/writing-marble-tests.md).
 
-    effects = TestBed.get&lt;MyEffects&gt;(MyEffects);
-  });
+The `hot`, `cold`, and `toBeObservable` functions are imported from [`jasmine-marbles`](https://www.npmjs.com/package/jasmine-marbles).
 
-  it('should work', () => {
-    actions = new ReplaySubject(1);
-    actions.next(MyActions.exampleAction());
+</div>
 
-    effects.someSource$.subscribe(result => {
-      expect(result).toEqual(MyActions.exampleActionSuccess());
-    });
-  });
+<code-example header="my.effects.spec.ts">
+// mock Actions to represent a user that is typing
+actions$ = hot('-a-b-', {
+  a: { type: 'SEARCH CUSTOMERS', name: 'J' },
+  b: { type: 'SEARCH CUSTOMERS', name: 'Jes' },
+})
+
+// mock the service to prevent an HTTP request to return an array of customers
+customersServiceSpy.searchCustomers.and.returnValue(
+  cold('--a|', { a: [...] })
+);
+
+// expect the first action to debounce and not to dispatch
+// expect the second action to result in a SUCCESS action
+const expected = hot('-------a', {
+  a: {
+    type: 'SEARCH CUSTOMERS SUCCESS',
+    customers: [...],
+  },
 });
-</code-example>
 
-### getEffectsMetadata
+expect(
+  effects.searchCustomers$({
+    debounce: 20,
+    scheduler: getTestScheduler(),
+  })
+).toBeObservable(expected);
+</code-example>
 
-Returns metadata configuration for all effects in a class instance.
-Use this function to ensure that effects have proper metadata attached.
+### With Observables
 
-Usage:
+To test simple Effects, it might be easier to create an Observable instead of using a marble diagram.
 
 <code-example header="my.effects.spec.ts">
-import { TestBed } from '@angular/core/testing';
-import { EffectsMetadata, getEffectsMetadata } from '@ngrx/effects';
-import { MyEffects } from './my-effects';
-
-describe('My Effects', () => {
-  let effects: MyEffects;
-  let metadata: EffectsMetadata&lt;MyEffects&gt;;
-
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [
-        MyEffects,
-        // other providers
-      ],
-    });
+// mock Actions and immediately dispatch a GET action
+actions$ = of({ type: 'GET CUSTOMERS' });
 
-    effects = TestBed.get&lt;MyEffects&gt;(MyEffects);
-    metadata = getEffectsMetadata(effects);
-  });
+// mock the service to prevent an HTTP request
+customersServiceSpy.getAllCustomers.and.returnValue(of([...]));
```

Comment:
IMHO the `[...]` might be confusing, maybe you could use `customerList` instead?